snapcraft/internal/elf.py

```diff
@@ -268,13 +266,18 @@ def __init__(self, *, path: str) -> None:
         self.needed = elf_data[3]
         self.execstack_set = elf_data[4]
         self.is_dynamic = elf_data[5]
+        self.build_id = elf_data[6]
```

Comment:
nit: add typing
```
self.build_id: str = elf_data[6]
self.has_debug_info: bool = elf_data[7]
```

But it is consistent with everything around it.

Reply:
Can mypy not infer the types from the `_extract` return type?

Reply:
Ah yes, silly me.  I'm already imaging the future ElfFile where everything is defined in init and not by "the" Tuple! :) Thanks!